utils/swift_build_support/swift_build_support/products/playgroundsupport.py

```diff
@@ -0,0 +1,131 @@
+# swift_build_support/products/playgroundsupport.py -------------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+import os
+import re
+from . import product
+from .. import shell
+from .. import targets
+
+
+class PlaygroundSupport(product.Product):
+    @classmethod
+    def product_source_name(cls):
+        return "swift-xcode-playground-support"
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    def should_build(self, host_target):
+        return self.args.build_playgroundsupport
+
+    def build(self, host_target):
+        root = os.path.dirname(os.path.dirname(self.toolchain.swiftc))
+        swift_lib_dir = os.path.join(root, 'lib', 'swift')
+        (host_os, host_arch) = host_target.split('-')
+
+        with shell.pushd(self.source_dir):
+            shell.call([
+                "xcodebuild",
+                "-configuration", self.args.build_variant,
+                "-workspace", "swift-xcode-playground-support.xcworkspace",
+                "-scheme", "BuildScript-{}".format({
```

Comment:
I feel like we should have a helper function of variable for this rather than compute it multiple times in-line.

Reply:
It is only computed twice (once in build, once in install).  It is a dictionary lookup, which is pretty fast.  I think that trying to extract this and reduce the duplication would actually hurt the readability of the build process.